src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -1495,6 +1497,161 @@ static bool extendedTypeIsOrDerivedFrom(NamedTypeSymbol hiddenContainer, NamedTy
             }
         }
 
+        /// <summary>
+        /// We remove non-methods that are "hidden" (based on extended type) by more specific members.
+        /// But we collect all the methods, because we'll only know if they are "hidden" after we have the complete set of extension members.
+        ///
+        /// Once we have the complete set of extension members, we'll call <see cref="CompleteMergeExtensionLookupResultsHidingLessSpecific(LookupResult, ConsList{TypeSymbol}, ref CompoundUseSiteInfo{AssemblySymbol})"/>
+        /// It will remove methods that are "hidden" by any remaining non-methods.
+        /// It will also compact the set (deal with any null entries left in the result).
+        /// </summary>
+        private void MergeExtensionLookupResultsHidingLessSpecificNonMethods(LookupResult current, LookupResult additional, ConsList<TypeSymbol> basesBeingResolved, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
+        {
+            if (!current.IsMultiViable || !additional.IsMultiViable)
+            {
+                current.MergePrioritized(additional);
```

Comment:
>current.MergePrioritized(additional); [](http://example.com/codeflow?start=16&length=37)

Do we have test covering this code path?

Reply:
Yes, `PreferMoreSpecific_Static_InvocableFieldAndMoreSpecificMethodAndNotViable`. The result from the empty extension type (`E3`) is empty and so it is not viable.

Reply:
Is it possible to get here with both results viable, but with at least one one not MultiViable?

Reply:
If both results are viable, then both of the are MultiViable.

```
   internal bool IsMultiViable
   {
       get
       {
           return Kind == LookupResultKind.Viable;
       }
   }
```